src/lib/uTop.ml

```diff
@@ -33,7 +39,12 @@ let version = "%%VERSION%%"
    +-----------------------------------------------------------------+ *)
 
 let history = LTerm_history.create []
-let history_file_name = ref (Some (Filename.concat LTerm_resources.home ".utop-history"))
+
+let unopt = function
+        | Some(v) -> v
+        | None -> "~/"
```

Comment:
I still believe this is not a good idea.

Here's a solution :

```ml
let history_file_name = ref (Option.map (fun cache_dir -> Filename.concat cache_dir "history") M.cache_dir)
```

(and remove your `unopt` function)

Reply:
But then again, what if M.cache_dir is None?

Reply:
It's [already handled](https://github.com/ocaml-community/utop/blob/560617c64f544dba7fe5b6f1a2ad1a963b3d36e4/src/lib/uTop_main.ml#L47-L49) and it won't happen in practice.

Reply:
Alright then.